### ISSUE TITLE
RuntimeError: dictionary changed size during iteration

### ISSUE THREAD

**Issue by sagarbansal88** [whamcloud/integrated-manager-for-lustre#1269]
While testing: IML Upgrade testing from 5.0.x managed ldiskfs to 5.1

Installed IML on the adm node using following:
vagrant provision adm --provision-with install-iml-5

On step "Select Managed Storage Server from the Select Server Profile drop down. Finally, click Proceed." it failed with error "Reboot of oss1.local required to switch from running kernel kernel-3.10.0-957.12.2.el7.x86_64 to required kernel-3.10.0-957.10.1.el7_lustre.x86_64"

Should this message be experienced by customers?
Also post this I didn't find option to even re-try on UI post reboot.

![image](https://user-images.githubusercontent.com/56718460/67080306-90f5a080-f1b2-11e9-9798-07a82dbe2250.png)

**Comment by sagarbansal88** [comment]
The action buttons are disabled for infinite time for all the servers.

**Comment by jgrund** [comment]
@sagarbansal88 Please attach `sosreport`s for the manager and agent nodes to this issue.

**Comment by jgrund** [comment]
> The action buttons are disabled for infinite time for all the servers.

This is https://github.com/whamcloud/GUI/pull/228, which should now be fixed. 

You can verify by doing the following on the adm node:

```
yum clean metadata
yum reinstall iml-gui
systemctl restart iml-manager.target
```

**Comment by jgrund** [comment]
Looking at the node, I see this:

```
[2019-10-16 11:56:21,132: INFO/job] StatefulObject.set_state oss1.local LNet configuration unconfigured->lnet_unloaded (intentional=True) 140388115839888
[2019-10-16 11:56:21,151: CRITICAL/job_scheduler] Unhandled exception in RunJobThread: Traceback (most recent call last):
  File "/usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py", line 282, in run
    self._run()
  File "/usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py", line 306, in _run
    self.job.id, step_klass=klass, args=clean_args, step_index=step_index, step_count=len(self.steps)
  File "/usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py", line 209, in getter
    self.put(deepcopy((name, args, kwargs)))
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 237, in _deepcopy_tuple
    y.append(deepcopy(a, memo))
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/usr/lib64/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/usr/lib64/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/usr/lib64/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/usr/lib64/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 257, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/usr/lib64/python2.7/copy.py", line 190, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/usr/lib64/python2.7/copy.py", line 334, in _reconstruct
    state = deepcopy(state, memo)
  File "/usr/lib64/python2.7/copy.py", line 163, in deepcopy
    y = copier(x, memo)
  File "/usr/lib64/python2.7/copy.py", line 256, in _deepcopy_dict
    for key, value in x.iteritems():
RuntimeError: dictionary changed size during iteration
```

**Comment by jgrund** [comment]
This is actually caused by one of the database records changing during a job. I'm reluctant to fix this since it's in 5.0 and can be worked around by restarting the manager service `chroma-config restart`

**Comment by sagarbansal88** [comment]
I tried the following and the buttons were active
yum clean metadata
yum reinstall iml-gui
systemctl restart iml-manager.target
